### PR TITLE
Support different base URIs

### PR DESCRIPTION
**Pull request by joneshf**
See #4.

I'm not entirely satisfied with this. It feels like the better way forward is to find out how to make it so orphan instances don't pop up. But, it seems to do what we discussed.

@owickstrom thoughts?

### CODE REVIEW

File: src/Type/Trout/Client.purs [hunk 1]
```diff
@@ -27,17 +33,38 @@ import Type.Trout.ContentType.JSON (JSON)
...
+prependSegment segment req =
+  req { path = singleton segment <> req.path }
+
 appendSegment :: String -> RequestBuilder -> RequestBuilder
 appendSegment segment req =
   req { path = req.path <> singleton segment }
 
 toAffjaxRequest :: RequestBuilder -> AffjaxRequest Unit
-toAffjaxRequest req = defaultRequest { url = "/" <> joinWith "/" req.path }
+toAffjaxRequest req = defaultRequest { url = joinWith "/" req.path }
```
Comment (owickstrom) [inline src/Type/Trout/Client.purs]: Doesn't this change mean that types without `BaseURI` will be relative? What about a `baseUri :: Maybe String` field in the `RequestBuilder`, or just a `String` defaulting to `/`?

Comment (joneshf) [reply src/Type/Trout/Client.purs]: Good idea. Adding a field sounds good.

**Comment by joneshf** [comment]
@owickstrom what do you think?

**Comment by owickstrom** [comment]
@joneshf Thanks for the reminder! Everything looks good, except that maybe the `BaseURI` should not assume that you always want to have just the scheme and authority parts, e.g. when you run the API locally on `http://localhost` and in production you have `https://example.com/api/the-cool-thing` as base URI. Maybe have `BaseURI` wrap an `AbsoluteURI`?

**Comment by nsaunders** [comment]
Resolved in #16

**Pull request closed without merging**